app/assets/javascripts/pretty-text/engines/discourse-markdown/emoji.js

```diff
@@ -224,6 +225,23 @@ function applyEmoji(
     });
   }
 
+  if (emojiDenyList) {
+    emojiDenyList.split("|").forEach((emoji) => {
+      let regex = new RegExp(`:${emoji}:`, "g");
+      content = content.replace(regex, "");
+
+      // remove any aliases of denied emoji
+      Object.entries(aliases).forEach(([key, list]) => {
```

Comment:
Related to my comment in https://github.com/discourse/discourse/pull/20929/files#r1157924786, if `emojiDenyList` already includes aliases, we will be able to greatly simplify the logic here.